modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -782,7 +883,10 @@ namespace cv {
                     if(activation == "leaky")
                         ++cv_layers_counter;  // For ReLU
 
-                    current_channels = net->out_channels_vec[darknet_layers_counter];
+                    if(!darknet_layers_counter)
```

Comment:
What this `if` condition means? I think that we need to do this only in case of `layer_type == "connected"`

Reply:
If Connected is the first layer of model, only then we need `total(tensor_shape)` i.e. net->channels * net->width * net->height. 
For all other cases, we can get input size in `tensor_shape[0]` from https://github.com/opencv/opencv/blob/24932c781ea872cf542536402793b43a8d8c0216/modules/dnn/src/darknet/darknet_io.cpp#L889
That's why after first layer, tensor_shape is resized to 1. otherwise, it will hurt here
https://github.com/opencv/opencv/blob/24932c781ea872cf542536402793b43a8d8c0216/modules/dnn/src/darknet/darknet_io.cpp#L838-L839